src/SFA.DAS.EAS.Web/FileDownloads/Payments/ExcelTransactionFormatter.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System.Collections.Generic;
+using SFA.DAS.EAS.Domain.Interfaces;
+using SFA.DAS.EAS.Domain.Models.Transaction;
+using SFA.DAS.EAS.Web.ViewModels;
+
+namespace SFA.DAS.EAS.Web.FileDownloads.Payments
+{
+    public class ExcelTransactionFormatter : ITransactionFormatter
+    {
+        private readonly IExcelService _excelService;
+
+        public ExcelTransactionFormatter(IExcelService excelService)
+        {
+            _excelService = excelService;
+        }
+
+        public byte[] GetFileData(List<TransactionDownloadLine> transactions)
+        {
+            var excelRows = new List<string[]>();
+
+            excelRows.Add(GetHeaderRow());
+
+            foreach (var transaction in transactions)
+            {
+                excelRows.Add(new[]
+                {
+                    transaction.DateCreated.ToString(), transaction.TransactionType,
+                    transaction.EmpRef, transaction.PeriodEnd, transaction.LevyDeclared.ToString(),
+                    transaction.EnglishFraction.ToString(), transaction.TenPercentTopUp.ToString(),
+                    transaction.TrainingProvider, transaction.CohortReference, transaction.Uln,
+                    transaction.Apprentice, transaction.ApprenticeTrainingCourse,
+                    transaction.PaidFromLevy.ToString(), transaction.EmployerContribution.ToString(),
+                    transaction.GovermentContribution.ToString(), transaction.Total.ToString()
+                });
+            }
+
+            var transactionData = new Dictionary<string, string[][]>
+            {
+                {"Transactions", excelRows.ToArray()}
+            };
+
+            var fileData = _excelService.CreateExcelFile(transactionData);
+
+            return fileData;
+        }
+
+        private string[] GetHeaderRow()
+        {
+            return new[]
```

Comment:
I don't thinks this array is modified? If not could it be made private static so that a new array is not created for each request?

Reply:
yep correct, forgot to move it. done.